### PR TITLE
AR Commands

### PR DESCRIPTION
**Pull request by t-constantin**
New ar commands:
- Scan an april tag to see commands linked to robot
- If robot doesnt have commands, refresh with upper right button
- changed command cards to have increments button
- not supporting non numerical input values. must be changed in command view. (keyboard issues)
- small fixes

### CODE REVIEW

File: HiveAR/app/src/main/java/com/swarmus/hivear/adapters/ARCommandsAdapter.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            increase.setOnClickListener(v -> {
+                if (arg.getArgumentType().equals(Integer.class)) {
+                    int value = Integer.parseInt(arg.getValue());
+                    argInput.setText(String.valueOf(value + 1));
+                }
+
+                if (arg.getArgumentType().equals(Float.class)) {
+                    float value = Float.parseFloat(arg.getValue());
+                    argInput.setText(String.valueOf(value + 0.5));
+                }
```
Comment (xgroleau) [inline HiveAR/app/src/main/java/com/swarmus/hivear/adapters/ARCommandsAdapter.java:108]: Maybe add an else and log an unsupported type, same for decresing

File: HiveAR/app/src/main/java/com/swarmus/hivear/adapters/ARCommandsAdapter.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        }
+        else {
+            function = buzzFunctionTemplateList.at(position);
+        }
+
+        holder.commandNameTV.setText(function.getName());
+        holder.commandNameTV.setCompoundDrawablesWithIntrinsicBounds(
+                0,
+                0,
+                function.isBuzzFunction() ? R.drawable.ic_buzz : 0,
```
Comment (xgroleau) [inline HiveAR/app/src/main/java/com/swarmus/hivear/adapters/ARCommandsAdapter.java:68]: What happens if a buzz function and a host function have the same name in the UI? Do they have a small icon next to the name or something to differentiate?

File: HiveAR/app/src/main/java/com/swarmus/hivear/fragments/ARViewFragment.java [hunk 3]
```diff
@@ -139,6 +151,21 @@ public void onDestroyView() {
...
+                e.printStackTrace();
+            }
+        }
+
+        if (agentBuzzCommandObserver != null && agent != null) {
+            try {
+                agent.deleteObserver(agentBuzzCommandObserver);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
```
Comment (xgroleau) [inline HiveAR/app/src/main/java/com/swarmus/hivear/fragments/ARViewFragment.java:167]: Is it common in java to just catch all the exceptions and not handling it?

**Review by xgroleau — APPROVED** [review]
Minor comments

File: HiveAR/app/src/main/java/com/swarmus/hivear/adapters/ARCommandsAdapter.java (hunk 2 above)
Comment (t-constantin) [reply HiveAR/app/src/main/java/com/swarmus/hivear/adapters/ARCommandsAdapter.java:68]: yes a little bug icon aside buzz commands :)

File: HiveAR/app/src/main/java/com/swarmus/hivear/fragments/ARViewFragment.java (hunk 3 above)
Comment (t-constantin) [reply HiveAR/app/src/main/java/com/swarmus/hivear/fragments/ARViewFragment.java]: unecessary try/catch, removed it.
Although for the question I don't know much!

File: HiveAR/app/src/main/java/com/swarmus/hivear/fragments/ARViewFragment.java [hunk 4]
```diff
@@ -421,7 +494,23 @@ private void initARIndicator(AnchorNode parent, Agent agent) {
...
+        }
+
+        if (agentBuzzCommandObserver != null && agent != null) {
+            try {
+                agent.deleteObserver(agentBuzzCommandObserver);
+            } catch (Exception e) {
+                e.printStackTrace();
```
Comment (xgroleau) [inline HiveAR/app/src/main/java/com/swarmus/hivear/fragments/ARViewFragment.java:509]: Then is this trycatch and the commandObserver one is necessary?

**Pull request merged**